### ISSUE TITLE
EndsWith should use rfind()

### ISSUE THREAD

**Issue by lichray** [philsquared/Catch#573]
https://github.com/philsquared/Catch/blob/7424b23bfb4423fd1213141633844f09faaff8bc/single_include/catch.hpp#L1089

`.find()` is used.

**Comment by philsquared** [comment]
Wow that's terrible code! Did I write that? (checks... yes I did. I am ashamed).
You are quite right that, given the approach being used it should be using `rfind()`.
But the approach itself seems like an awkward way to do it. I can only imagine that I started with `Contains()` and made the minimum changes necessary to morph it into `StartsWith()`, then `EndsWith()`.
But given that I already had the functions `startsWith()` and `endsWith()` (https://github.com/philsquared/Catch/blob/master/include/internal/catch_common.hpp#L19) I don't understand why I didn't use them.

So, thanks for bringing this up! I'll take a closer look later and clean things up a bit (fixing the latent bug in the process).
